lib/pure/strutils.nim

```diff
@@ -154,6 +154,54 @@ proc isDigit*(c: char): bool {.noSideEffect,
     doAssert isDigit('8') == true
   return c in Digits
 
+proc isNumeric*(s: string): bool =
+  ## Checks whether the string is numeric.
+  ## When the string is an integer, float or exponential, it returns true,
+  ## otherwise it returns false.
+  ##
+  ## **Note:** The reason that `parseFloat()` is not used to achieve this is its
+  ## poor performance.
+  runnableExamples:
+    doAssert isNumeric("123") == true
+    doAssert isNumeric("123.45") == true
+    doAssert isNumeric("+123.45") == true
+    doAssert isNumeric("123.45e-2") == true
+    doAssert isNumeric("+123.45E-2") == true
+    doAssert isNumeric("-123.45e2") == true
+    doAssert isNumeric("e123.45") == false
+    doAssert isNumeric("123abc") == false
+    doAssert isNumeric("123.45.6") == false
+    doAssert isNumeric("123.45e++5") == false
```

Comment:
```suggestion
    doAssert isNumeric("123.45e++5") == false
    doAssert isNumeric("Inf")
    doAssert isNumeric("+Inf")
    doAssert isNumeric("NaN")
```

- https://play.nim-lang.org/#ix=2DFU

I think `isNumeric` can be `func`, even with `--experimental:strictFuncs` compiles.

Reply:
IMO `doAssert isNumeric(...) == false` is not idiom in Nim world. It should be `doAssert not isNumeric(...)`